libs/fetcher.js

```diff
@@ -283,11 +283,27 @@ Fetcher.registerFetcher = function (fetcher) {
  * @param {Function} service
  */
 Fetcher.registerService = function (fetcher) {
-    if (!fetcher || !fetcher.name) {
-        throw new Error('Service is not defined correctly');
+    if (!fetcher) {
+        throw new Error(
+            'Fetcher.registerService requires a fetcher definition (ex. registerService(fetcher))'
+        );
     }
-    Fetcher.services[fetcher.name] = fetcher;
-    debug('fetcher ' + fetcher.name + ' added');
+
+    var resourceName;
+    if (typeof fetcher.resource !== 'undefined') {
+        resourceName = fetcher.resource;
+    } else if (typeof fetcher.name !== 'undefined') {
+        if ('production' !== process.env.NODE_ENV) {
+            console.warn('"fetcher.name" is deprecated. Please rename it to "fetcher.resource".');
```

Comment:
This will be spammy in the console since we have many fetchers that use name. if we could only deprecate it once when the server starts, that would be better.

Reply:
I've moved this logic to `Fetcher.middleware`. I also added an array to control all the deprecated services definitions. In this way, it will be easier to spot services non compliant with the newer API.